pycsob/client.py

```diff
@@ -17,6 +21,213 @@ def send(self, request, **kwargs):
+        return OrderedDict(data)
+
+
+@dataclass
+class CartItem(ConvertMixin):
+    """Cart item for creating card payment."""
+    # Documentation: https://github.com/csob/paymentgateway/wiki/Basic-methods#item---cart-item-object-cart-
+
+    name: str
+    quantity: int
+    amount: int
+    description: Optional[str] = None
+
+    _max_length = {"name": 20, "description": 40}
+
+
+@dataclass
+class CustomerAccount(ConvertMixin):
+    """Customer account data."""
+    # Documentation: https://github.com/csob/paymentgateway/wiki/Purchase-metadata#customeraccount-data-
+
+    created_at: Optional[datetime] = None
+    changed_at: Optional[datetime] = None
+    changed_pwd_at: Optional[datetime] = None
+    order_history: Optional[int] = None
+    payments_day: Optional[int] = None
+    payments_year: Optional[int] = None
+    oneclick_adds: Optional[int] = None
+    suspicious: Optional[bool] = None
+
+
+@unique
+class CustomerLoginType(Enum):
+    """Type of customer login."""
+    # Documentation: https://github.com/csob/paymentgateway/wiki/Purchase-metadata#customerlogin-data-
+
+    GUEST = "guest"
+    ACCOUNT = "account"
+    FEDERATED = "federated"
+    ISSUER = "issuer"
+    THIRDPARTY = "thirdparty"
+    FIDO = "fido"
+    FIDO_SIGNED = "fido_signed"
+    API = "api"
+
+
+@dataclass
+class CustomerLogin(ConvertMixin):
+    """Customer login data."""
+    # Documentation: https://github.com/csob/paymentgateway/wiki/Purchase-metadata#customerlogin-data-
+
+    auth: Optional[CustomerLoginType] = None
+    auth_at: Optional[datetime] = None
+    auth_data: Optional[str] = None
+
+
+@dataclass
+class Customer(ConvertMixin):
+    """Customer data for creating card payment."""
+    # Documentation: https://github.com/csob/paymentgateway/wiki/Purchase-metadata#customer-data-
+
+    name: str
+    email: Optional[str] = None
+    home_phone: Optional[str] = None
+    work_phone: Optional[str] = None
+    mobile_phone: Optional[str] = None
+    account: Optional[CustomerAccount] = None
+    login: Optional[CustomerLogin] = None
+
+    _max_length = {"name": 45, "description": 100}
+
+
+@dataclass
+class OrderAddress(ConvertMixin):
+    """Order address (billing or shipping)."""
+    # Documentation: https://github.com/csob/paymentgateway/wiki/Purchase-metadata#orderaddress-data-
+
+    address_1: str
+    city: str
+    zip: str
+    country: str
+    address_2: Optional[str] = None
+    address_3: Optional[str] = None
+    state: Optional[str] = None
+
+    _max_length = {"address_1": 50, "address_2": 50, "address_3": 50, "city": 50, "zip": 16}
+    _ordered_fields = ("address_1", "address_2", "address_3", "city", "zip", "country", "state")
+
+
+@dataclass
+class OrderGiftcards(ConvertMixin):
+    """Order giftcards data."""
+    # Documentation: https://github.com/csob/paymentgateway/wiki/Purchase-metadata#ordergiftcards-data-
+
+    total_amount: Optional[int] = None
+    currency: Optional[str] = None
```

Comment:
According to the documentation, it should be an enumeration type..

Reply:
Enumeration added.